examples/dmm/dmm.py

```diff
@@ -250,16 +254,22 @@ def guide(self, mini_batch, mini_batch_reversed, mini_batch_mask,
                 # to yield a transformed distribution that we use for q(z_t|...)
                 if len(self.iafs) > 0:
                     z_dist = TransformedDistribution(dist.Normal(z_loc, z_scale), self.iafs)
+                    assert z_dist.event_shape == (self.z_q_0.size(0),)
+                    assert z_dist.batch_shape == (len(mini_batch),)
                 else:
                     z_dist = dist.Normal(z_loc, z_scale)
-                assert z_dist.event_shape == ()
-                assert z_dist.batch_shape == (len(mini_batch), self.z_q_0.size(0))
+                    assert z_dist.event_shape == ()
+                    assert z_dist.batch_shape == (len(mini_batch), self.z_q_0.size(0))
 
                 # sample z_t from the distribution z_dist
                 with pyro.poutine.scale(scale=annealing_factor):
-                    z_t = pyro.sample("z_%d" % t,
-                                      z_dist.mask(mini_batch_mask[:, t - 1:t])
-                                            .to_event(1))
+                    if len(self.iafs) > 0:
```

Comment:
lgtm but can we add a comment about the presence/absence of `to_event` in the two cases?

Reply:
Okay, I'd added a few comments here:

`# in output of normalizing flow, all dimensions are correlated (event shape is not empty)`
`...`
`# when no normalizing flow used, ".to_event(1)" indicates latent dimensions are independent`